backend/src/middleware/auth.go

```diff
@@ -0,0 +1,76 @@
+package middleware
+
+import (
+	"github.com/GenerateNU/sac/backend/src/auth"
+	"github.com/GenerateNU/sac/backend/src/models"
+	"github.com/GenerateNU/sac/backend/src/types"
+
+	"github.com/gofiber/fiber/v2"
+)
+
+const (
+	AuthLoginPath = "/api/v1/users/auth/login"
+	AuthRefreshPath = "/api/v1/users/auth/refresh"
+	AuthLogoutPath = "/api/v1/users/auth/logout"
+)
+
+func Authenticate(c *fiber.Ctx) error {
+	// TODO: use a contains function instead of this
+	if c.Path() == AuthLoginPath || c.Path() == AuthRefreshPath || c.Path() == AuthLogoutPath {
```

Comment:
Register should likely be apart of the this list

Reply:
and logged out in a call you should only make if logged in so this should also probably be a protected path